File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        for(int width = 3; width <= 5000; width++){
            for(int height = 3; height <= width; height++){
                //조건 검사
                int boundary = (width + height - 2) * 2;
                int center = width * height - boundary;
                if(brown == boundary && yellow == center){
                    return new int[]{width, height};
                }
            }
        }
        return null;
    }
}
/*
규칙찾기
문제를 잘 읽어보고 문제에 해당하는 규칙을 찾기

*/

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    for i in range(brown // 2):
        for j in range(brown // 2):
            if 2*i + 2*j - 4 == brown and (i - 2) * (j - 2) == yellow:
                return [j, i]
"""
# 테두리 1줄은 갈색
# 중앙 노란색
# brown = 2*height + 2*width - 4
# yellow = (height - 2) * (width - 2)
# 가로가 세로보다 크거나 같다
갈색 최대 5000
brown

"""